daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs

```diff
@@ -672,23 +680,55 @@ runStart sandboxPort (StartNavigator shouldStartNavigator) (OpenBrowser shouldOp
                   then withNavigator
                   else (\_ _ _ f -> f sandboxPh)
 
-runDeploy :: String -> SandboxPort -> IO ()
-runDeploy host sandboxPort = do
+data HostAndPortFlags = HostAndPortFlags { hostM :: Maybe String, portM :: Maybe Int }
+
+withHostAndPortDefaults :: HostAndPortFlags -> IO HostAndPort
+withHostAndPortDefaults HostAndPortFlags{hostM,portM} = do
+    host <- fromMaybeM getProjectLedgerHost hostM
+    port <- fromMaybeM getProjectLedgerPort portM
+    return HostAndPort {host,port}
+
+runListParties :: HostAndPortFlags -> IO ()
+runListParties flags = do
+    hp <- withHostAndPortDefaults flags
+    putStrLn $ "Listing parties at " <> show hp
+    xs <- Ledger.listParties hp
+    when (null xs) $ putStrLn "no parties are known"
```

Comment:
I think
```suggestion
    if null xs then putStrLn "no parties are known" else mapM_ print xs
```
and removing the next line is clearer.

Reply:
man, we are getting dangerously close to the bike shed here, but I will do it...